docs/manual.md

```diff
@@ -158,7 +158,36 @@ A budget that equals the period (aka. a "full" budget) behaves like a traditiona
 The **priority** determines which of the runnable PDs to schedule. A PD is runnable if one of its entry points have been invoked and it has budget remaining in the current period.
 Runnable PDs of the same priority are scheduled in a round-robin manner.
 
-The **passive** determines whether the PD is passive. A passive PD will have it's scheduling context revoked after initialisation and then bound instead to the PD's notification object. This means the PD will be scheduled on receiving a notification, whereby it will run on the notification's scheduling context, or when the PD receives a *protected procedure* by another PD, whereby the passive PD will run on the scheduling context of the callee. 
+The **passive** determines whether the PD is passive. A passive PD will have it's scheduling context revoked after initialisation and then bound instead to the PD's notification object. This means the PD will be scheduled on receiving a notification, whereby it will run on the notification's scheduling context, or when the PD receives a *protected procedure* by another PD, whereby the passive PD will run on the scheduling context of the callee.
+
+If a PD runs out of its budget, it is preempted and resumed the next time period. An example of a task like that is below - the `for` loop will be preempted several times, depending on your budget:
+
+```C
+void
+notified(sel4cp_channel ch)
+{
+	switch (ch) {
+		case START_BIG_TASK: {
+				sel4cp_dbg_puts("Starting a big task\n");
+				int i = 0;
+				for (i = 0; i<100000000; i++) {
+					if (i % 10000000 == 0) {
+						sel4cp_dbg_puts(".");
+					}
+					asm("nop");
+				}
+				sel4cp_dbg_puts("Big task done\n");
+				break;
+			}
+		default: {
+			sel4cp_dbg_puts("received notification on unexpected channel\n");
+			break;
+			}
+		}
+}
+```
+
+On the other hand, if a PD has `budget` available, but receives no notification, it is not run at all. Returning from `notified()` function and calling `seL4_Yield()` during the PD execution has the same effect, and donates the remaining budget of the PD.
```

Comment:
"Returning from `notified()` function and calling `seL4_Yield()` during the PD execution has the same effect, and donates the remaining budget of the PD." Has the same effect as what exactly? I'm not sure what you mean by calling `seL4_Yield` after returning from the notified entry point?

Reply:
This is based on the discussion [here](https://github.com/au-ts/camkes_to_sel4cp_guide/issues/1#issuecomment-1507922309). In my understanding, you can either return from `notified()` or call `seL4_Yield()` somewhere in your `notified()` function, and that will end the execution of that particular PD in the given period. Is that accurate?

Reply:
While seL4CP doesn't stop anyone from doing this in their user-code, I am kind of hesitant to have this in the manual as I don't think there should be any need to call `seL4_Yield` and having a non-event driven PD seems to go against the principles of seL4CP.